src/transformState.js

```diff
@@ -5,7 +5,27 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const key of actions) {
+    switch (key.type) {
+      case 'addProperties':
+        Object.assign(state, key.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const value of key.keysToRemove) {
+          delete state[value];
+        }
+        break;
+
+      case 'clear':
+        for (const value in state) {
+          delete state[value];
+        }
+        break;
+    }
+  }
```

Comment:
destructure action object

Reply:
you do not use destructuring
for example
```
  const {
    type,
    extraData,
    keysToRemove,
  } = action;
```